src/dash-table/derived/style/index.ts

```diff
@@ -125,4 +127,20 @@ export const derivedTableStyle = memoizeOneFactory(
         convertStyle(defaultTable),
         convertStyle(table)
     ]
-);
\ No newline at end of file
+);
+
+export function resolveStyle(styles: IConvertedStyle[]): CSSProperties {
+    return R.mergeAll(
+        R.map<IConvertedStyle, CSSProperties>(
+            s => R.omit(BORDER_PROPERTIES_AND_FRAGMENTS, s.style),
+            styles
+        )
+    );
+}
+
+export const getDataCellStyle = (datum: Datum, i: number, column: IVisibleColumn) => R.compose(resolveStyle, matchesDataCell(datum, i, column));
+export const getDataOpCellStyle = (datum: Datum, i: number) => R.compose(resolveStyle, matchesDataOpCell(datum, i));
+export const getFilterCellStyle = (column: IVisibleColumn) => R.compose(resolveStyle, matchesFilterCell(column));
+export const getFilterOpCellStyle = () => R.compose(resolveStyle, getFilterOpStyles);
+export const getHeaderCellStyle = (i: number, column: IVisibleColumn) => R.compose(resolveStyle, matchesHeaderCell(i, column));
+export const getHeaderOpCellStyle = (i: number) => R.compose(resolveStyle, getHeaderOpStyles(i));
```

Comment:
Define composed functions for how to filter and reduce styles for various usage scenarios (data, dataOp, filters, etc.)